api/models.py

```diff
@@ -206,3 +206,45 @@ async def wait_for_node(cls, timeout):
         if timeout > current_time:
             time_delta = timeout - current_time
             await asyncio.sleep(time_delta.total_seconds())
+
+
+class RegressionData(BaseModel):
+    """Regression object details model"""""
+
+    class Config:
+        """Configuration attributes for RegressionData"""
+        use_enum_values = True
+
+    revision: Revision = Field(
+        description='Git revision of regression object'
+    )
+    status: StatusValues = Field(
+        description='Regression object status'
+    )
+
+
+class Regression(Node):
+    """API model for regression tracking"""
+
+    regression_data: List[RegressionData] = Field(
+        description='Regression details'
+    )
+
+    def __init__(self, **data):
+        if data.get('kind') is None:
+            data['kind'] = 'regression'
```

Comment:
Would that work?
```suggestion
    def __init__(self, kind='regression', **data):
```

Reply:
If we want to use that signature the function becomes:
```
def __init__(self, kind='regression', **data):
        if data.get('kind') is None:
            data['kind'] = kind
        super().__init__(**data)
```
I don't see much difference. But I think this one is more readable and simplified.

Reply:
WIth `kind='regression'` you know 'kind' will not be in `**data` because that's the keyword arguments. So this should work then:
```python
def __init__(self, kind='regression', **data):
        super().__init__(kind=kind, **data)
```

Reply:
Yes.